arch/arm/soc/nordic_nrf5/nrf52/soc.h

```diff
@@ -21,7 +21,13 @@
 
 #endif /* !_ASMLANGUAGE */
 
-#define FLASH_PAGE_ERASE_MAX_TIME_US 89700UL
-#define FLASH_PAGE_MAX_CNT 256UL
+#define FLASH_PAGE_ERASE_MAX_TIME_US 	89700UL
+#define FLASH_PAGE_MAX_CNT		256UL
+
+/* For IMG_MANAGER  */
+#if defined(CONFIG_SOC_FLASH_NRF5)
+#define FLASH_DRIVER_NAME		CONFIG_SOC_FLASH_NRF5_DEV_NAME
+#define FLASH_ALIGN			4
```

Comment:
So far for nRF52840 DTS item [write-block-size](https://github.com/zephyrproject-rtos/zephyr/blob/master/dts/arm/nordic/nrf52840.dtsi#L26) was added. I recommend to add similar path for all of nrf5 devices and use it here.

Reply:
I prepared two commits for update nrf52832 and nrf51822 DT. You can take them from my fork: https://github.com/nvlsianpu/zephyr/tree/flash_alignment_nrf5x

Reply:
not sure what you want me to do, can you apply this after we merge this PR?

Reply:
I would just replace this by, and add last two commits form my branch:
`#define FLASH_ALIGN	 FLASH_WRITE_BLOCK_SIZE`

In case you merege this as is (LGTM) I will apply a patch later.

Reply:
ok, done